avx2/util.c

```diff
@@ -0,0 +1,21 @@
+#include <stdint.h>
+#include "util.h"
+
+/*************************************************
+* Name:        zeroize
+*
+* Description: Zeroizes a block of memory
+*
+* Arguments:   - void *data: pointer to a block of memory to
+                            be zeroized
+*              - size_t size: size of memory ad \ref data
+**************************************************/
+void zeroize(void *data, size_t size) {
```

Comment:
If you look at libsodium's memzero they utilize C11's guaranteed safe memset or, because VS doesn't support C11, they use Window's guaranteed safe function. There's others in the long if else including `explicit_bzero`. I think an approach like that would be better.

https://en.cppreference.com/w/c/string/byte/memset
https://github.com/jedisct1/libsodium/blob/master/src/libsodium/sodium/utils.c#L126